### ISSUE TITLE
Add Colemak layout

### ISSUE THREAD

**Issue by dircsem** [Julow/Unexpected-Keyboard#129]
I've added this issue to suggest to add this layout to the app.
I'm trying to do that.

**Comment by dircsem** [comment]
I'm having this issue on building the app
not sure if this is related to the version of the 
Import: Wrong number of arguments
Called from...
/Unexpected-Keyboard/srcs/special_font/build.pe: line 10
make: *** [Makefile:105: _build/assets/special_font.ttf] Error 1

Is this related to some dependency?

**Comment by Julow** [comment]
Thanks for contributing!

This is due to the Fontforge dependency, you need at least the version `20201107`, which isn't available in many distros.

If you are ok with using Nix, you should have everything right with `nix-shell ./shell.nix`. As a last recourse, you can let the CI build the app for you (push a branch to your fork and go to the "Actions" tab on Github).
This is not ideal! I plan on removing the dependency on Fontforge soon.

**Comment by dircsem** [comment]
I  don't know how to use this nix-shell will give a try. 
It's good to have this instructions at the contributing doc, I'll add it there

**Comment by dircsem** [comment]
![photo_2022-04-24_15-24-26](https://user-images.githubusercontent.com/2041058/164990848-401847a4-2629-48e1-a735-4ba1b29baf90.jpg)
Any tips on how to position the non character keys? Is there a problem to split esc and shift? Is is recommended to have symbol keys apart from the alphabet keys, or the idea is to have those always placed with alphabet keys?

**Comment by Julow** [comment]
There's no constraint on the placement of shift and backspace, the dvorak layout have them on the top row too.

It's also entirely fine to have keys for non-alphabet characters like Esc, `>`, `<`, Tab, etc..., the other layouts are like this mainly to look more familiar. However, you might end up with very tiny keys: Shift and Backspace are on the third row for that reason.

You can also consider modifying the bottom row by adding `bottom_row="false"` to the `<keyboard>` markup and taking the bottom row from `bottom_row.xml`. The neo2 layout does that: e52e537

Btw, Fontforge is no longer a build dependency (39952f8).

**Comment by dircsem** [comment]
![photo_2022-04-24_20-15-39](https://user-images.githubusercontent.com/2041058/165000745-3386b5cf-a5f4-474d-8f08-322642851f28.jpg)
I think that the keys are ok for now, I just need to adapt the symbols.

**Comment by dircsem** [comment]
![photo_2022-04-24_20-41-32](https://user-images.githubusercontent.com/2041058/165001565-a90c2033-2129-468d-8935-1af1c4d6124b.jpg)
Appreciate any comments, I`ve tryed to follow my keyboard for the symbols (it's not a colemak) There were some symbols (accents) that I've put but it didn't appeared I take the dvorak as base.
Should I maintain the dvorak symbols at the same place? Is there such a standard?

**Comment by Julow** [comment]
Sorry for the slow reply. The layouts looks good to me, have you had time to use it ?

The Esc and Tab keys do look a bit inconsistent but I'm not against. There's a few characters near the edges of the screen (`0`, `.`, `@`) that could be difficult to type for some users, it would be better to move some of them but that's not a strong requirement.

The placement of accents doesn't need to follow any standard, I usually place them not too far from the key they would often be used with but the placement is becoming meaningless as more languages are added. No policy have been enforced on contributed layouts so this is almost random at this point.
In the future, I plan on placing the accents automatically, layouts would no longer need to reference them.

**Comment by dircsem** [comment]
I'm using it since last week. 
I was thinking something similar, like my ~ is near a, and in Portuguese this is one of the places that we most use it (but I don't know for other languages) 
I was thinking on place cedilla near C and also below the key.
My plan is to move á to the right of the key, and put à at the same key but on the left side of it. Also thought to put ^ accent near e key.
My difficulties now is the < key that I sometimes press it when trying to press m, not sure how to handle it, because I can move the  backspace to there, but this won't solve my problem.
Why do you think that esc and tab are inconsistent, because of their size?

**Comment by dircsem** [comment]
About the 0 I tried to maintain all the numbers at the same line and I didn't want to place a number at the esc key, but I think that phones that this can be a problem for phones who have a curved screen, there are some spaces where I can move @ and . For me it seems anti intuitive to have keys on the left of the key on the left border, this also happens when we have right on right border, but I'm not sure if this is because I'm not used to it.
I'm missing an auto uppercase when I end a phrase, after a dot or at the first letter of a textfield.

**Comment by Julow** [comment]
If you think the layout is ready, you can send a PR.

>  Why do you think that esc and tab are inconsistent, because of their size?

All other layouts have Tab and Esc in the corners of other keys. An other thing that is inconsistent is the number of keys on each rows. I'm not against but I'd appreciate if you could try other ways.

>  I'm missing an auto uppercase when I end a phrase, after a dot or at the first letter of a textfield.

I plan on adding this feature at some point.

**Comment by dircsem** [comment]
I've just seen the querty and ok I'll change that.
I'll follow your standard for other keys. I understand that you've put the 0 on the left side of the key, but I think that it's unusual to have all numbers on the right and just zero on the left.
By the inconsistent number of the keys in each row, changing esc and tab can fix that. I don't like putting delete besides M because I always accidentally click there, not sure if I can put the key but with an space between them I'll test that to see how it will behave.

**Comment by Julow** [comment]
> but I think that it's unusual to have all numbers on the right and just zero on the left.

Agreed. It was a hard decision to make, I don't mind if this layout doesn't follow it.

> I don't like putting delete besides M

That's fine!

**Comment by dircsem** [comment]
f11_placeholder and some other accents are not working here, is this related to my language config?

**Comment by dircsem** [comment]
![photo_2022-05-01_20-31-04](https://user-images.githubusercontent.com/2041058/166169021-0ff3d407-3ede-451c-8111-c41c84b6c8d4.jpg)
I'll test this layout for a week, I think that it's good.

**Comment by Julow** [comment]
https://github.com/Julow/Unexpected-Keyboard/pull/135 is merged. Thanks a lot!

**Comment by dircsem** [comment]
Great thanks, I was off too.
I've decided to put math operations close to each other, but I found that - and _ should be close otherwise it's hard to find those symbols.
I still not comfortable with the accents position I'm testing some changes and as soon as I got a better layout I'll post here.
